hibernate-core/src/main/java/org/hibernate/dialect/pagination/LimitHandler.java

```diff
@@ -41,10 +43,16 @@ public interface LimitHandler {
 
 	String processSql(String sql, Limit limit);
 
+	@Deprecated // Never called directly by Hibernate ORM
 	default String processSql(String sql, Limit limit, QueryOptions queryOptions) {
 		return processSql( sql, limit );
 	}
 
+	// This is the one called directly by Hibernate ORM
+	default String processSql(String sql, int jdbcParameterBindingsCnt, @Nullable ParameterMarkerStrategy parameterMarkerStrategy, Limit limit, QueryOptions queryOptions) {
```

Comment:
I don't see a reason to make this nullable.

```suggestion
	default String processSql(String sql, int jdbcParameterBindingsCnt, ParameterMarkerStrategy parameterMarkerStrategy, Limit limit, QueryOptions queryOptions) {
```